x/superfluid/keeper/concentrated_liquidity_test.go

```diff
@@ -0,0 +1,247 @@
+package keeper_test
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+	cltypes "github.com/osmosis-labs/osmosis/v15/x/concentrated-liquidity/types"
+	"github.com/osmosis-labs/osmosis/v15/x/superfluid/keeper"
+	"github.com/osmosis-labs/osmosis/v15/x/superfluid/types"
+)
+
+func (suite *KeeperTestSuite) TestAddToConcentratedLiquiditySuperfluidPosition() {
+	defaultJoinTime := suite.Ctx.BlockTime()
+	type sendTest struct {
+		superfluidDelegated    bool
+		superfluidUndelegating bool
+		unlocking              bool
+		overwritePositionId    bool
+		amount0Added           sdk.Int
+		amount1Added           sdk.Int
+		doNotFundAcc           bool
+		isLastPositionInPool   bool
+		expectedError          error
+	}
+	testCases := map[string]sendTest{
+		"add to position that is superfluid delegated, not unlocking": {
+			superfluidDelegated: true,
+			amount0Added:        sdk.NewInt(100000000),
+			amount1Added:        sdk.NewInt(100000000),
+		},
+		"add to position that is superfluid delegated, not unlocking, not enough funds to add": {
```

Comment:
Please add a test cases where non-owner executes this

Reply:
non owner test case added https://github.com/osmosis-labs/osmosis/pull/5042/commits/f2e4b9501ddebe8cfb314a01dddea47cbe8c513e